swift/Sources/SignalClient/SenderKeyName.swift

```diff
@@ -6,23 +6,24 @@
 import SignalFfi
 
 public class SenderKeyName: ClonableHandleOwner {
-    internal override class func destroyNativeHandle(_ handle: OpaquePointer) {
-        signal_sender_key_name_destroy(handle)
+    internal override class func destroyNativeHandle(_ handle: OpaquePointer) -> SignalFfiErrorRef? {
+        return signal_sender_key_name_destroy(handle)
     }
 
     internal override class func cloneNativeHandle(_ newHandle: inout OpaquePointer?, currentHandle: OpaquePointer?) -> SignalFfiErrorRef? {
         return signal_sender_key_name_clone(&newHandle, currentHandle)
     }
 
+    // FIXME: Do we need both of these entry points?
```

Comment:
@jack-signal?

Reply:
Not necessarily needed I guess. Depends on what is going to be convenient for iOS. My take would be to leave this one then have the version taking a `ProtocolAddress` call `self.init` so underlying logic isn't duplicated, because if some name isn't already in a `ProtocolAddress`, creating one involves several new heap allocations which would be otherwise completely avoidable.

Reply:
(I don't have strong feelings about this and if it turns out that one form or another would be more convenient when we actually get to this in iOS it would be easy to just add it back)

Reply:
Hm, I like types but I also like avoiding heap allocations! I suppose another option is to make ProtocolAddress an eagerly-converted type like Fingerprint, since we're pretty sure it'll never grow another field or something. (But that does make it more awkward to validate the name.)

I'll go with your suggestion for now.